scripts/tier_analyzer.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+
+import argparse
+import re
+import subprocess
+import sys
+
+
+def check_virtualenv():
+    """Check if we are in virtualenv and if not, exit."""
+    if not hasattr(sys, 'real_prefix'):
+        print('You must activate CFME virtualenv in oder to run this script.')
+        sys.exit(1)
+
+
+def get_pytest_output(args, use_tier_marker=False):
+    """Get string output of pytest command."""
+    print('Getting pytest output {} tier marker.'.format('with' if use_tier_marker else 'without'))
+    pytest_args = ['pytest',
+        args.test_path,
+        '--collect-only',
+        '--long-running',
+        '--use-provider',
+        args.provider,
+        '-m']
+    if use_tier_marker:
+        pytest_args.append(args.tier_marker)
+    else:
+        pytest_args.append('uses_testgen')
+
+    return subprocess.check_output(pytest_args)
+
+
+def parse_pytest_output(output):
+    """Parse raw pytest output to get list of parametrized test cases."""
+    test_cases = []
+    tc_regex = re.compile(r"\s*<Function '(?P<test_case_name>test_\w+)(?P<parameters>\[.+\])?'>")
```

Comment:
**OPTIONAL:** Consider changing tc_regex to testcase_regex.   It will be much more apparent what it is and it won't make any lines too long.